python/ray/serve/drivers.py

```diff
@@ -174,3 +180,77 @@ async def get_intermediate_object_refs(self) -> Dict[str, Any]:
     async def get_dag_node_json(self) -> str:
         """Returns the json serialized root dag node"""
         return self.dags[self.MATCH_ALL_ROUTE_PREFIX].dag_node_json
+
+
+@PublicAPI(stability="alpha")
+class gRPCIngress:
+    """
+    gRPC Ingress that starts gRPC server based on the port
+    """
+
+    def __init__(self, port=DEFAULT_GRPC_PORT):
+        self.server = grpc.aio.server()
+        self.port = port
+
+        def register_servicer():
+            """
+            protobuf Schema gRPC should generate bind function
+            (e.g. add_PredictAPIsServiceServicer_to_server) to bind gRPC server
+            and schema interface
+            """
+            bind_function_name = "add_{}_to_server"
+            module_name = self.__class__.__bases__[1].__module__
+            servicer_name = self.__class__.__bases__[1].__name__
+            getattr(sys.modules[module_name], bind_function_name.format(servicer_name))(
+                self, self.server
+            )
+
+        register_servicer()
+
+        self.setup_complete = asyncio.Event()
+        self.running_task = asyncio.get_event_loop().create_task(self.run())
+
+    async def run(self):
+        self.server.add_insecure_port("[::]:{}".format(self.port))
+        self.setup_complete.set()
+        await self.server.start()
+        await self.server.wait_for_termination()
+
+
+if not isinstance(serve_pb2_grpc.PredictAPIsServiceServicer, type):
+    # The reason we have this fake Driver class is that sphinx-build is to mock
+    # PredictAPIsServiceServicer and it will cause the metaclass conflict issue
+    # from multiple inheritance.
+    @serve.deployment
+    class gRPCDriver:
+        def __init__(self):
+            raise Exception("gRPC driver Placeholder, it is not expected to be used")
+
+else:
+
+    @serve.deployment(driver_deployment=True, ray_actor_options={"num_cpus": 0})
+    class gRPCDriver(gRPCIngress, serve_pb2_grpc.PredictAPIsServiceServicer):
```

Comment:
should we name this something like DefaultgRPCDriver? or something else? current term is too generic and confused with gRPCIngress